web/jsonb/jsonbbasics/src/main/java/javaeetutorial/jsonbmodel/JsonbBean.java

```diff
@@ -0,0 +1,158 @@
+    
+    /* Form properties */
+    protected static final String PHONE_TYPE_HOME = "Home";
+    protected static final String PHONE_TYPE_WORK = "Work";
+
+    private String name = "Jason Bourne";
+    private String profession = "Super agent";
+
+    private String phoneType1 = PHONE_TYPE_HOME;
+    private String number1 = "123-456-789";
+
+    private String phoneType2 = PHONE_TYPE_WORK;
+    private String number2 = "123-555-555";
+
+    protected String jsonTextArea = "";
+    
+    public JsonbBean() {}
+    
+    /* Getters and setters */
+    public String getPhoneTypeHome() {
+        return PHONE_TYPE_HOME;
+    }
+    public String getPhoneTypeWork() {
+        return PHONE_TYPE_WORK;
+    }
+    public String getName() {
+        return name;
+    }
+    public void setName(String name) {
+        this.name = name;
+    }
+    public String getProfession() {
+        return profession;
+    }
+    public void setProfession(String profession) {
+        this.profession = profession;
+    }
+    public String getPhoneType1() {
+        return phoneType1;
+    }
+    public void setPhoneType1(String phoneType1) {
+        this.phoneType1 = phoneType1;
+    }
+    public String getNumber1() {
+        return number1;
+    }
+    public void setNumber1(String number1) {
+        this.number1 = number1;
+    }
+    public String getPhoneType2() {
+        return phoneType2;
+    }
+    public void setPhoneType2(String phoneType2) {
+        this.phoneType2 = phoneType2;
+    }
+    public String getNumber2() {
+        return number2;
+    }
+    public void setNumber2(String number2) {
+        this.number2 = number2;
+    }
+    public String getJsonTextArea() {
+        return jsonTextArea;
+    }
+    public void setJsonTextArea(String jsonTextArea) {
+        this.jsonTextArea = jsonTextArea;
+    }
+    
+    /* Action method for the form in index.xhtml.
+     * Creates Person object and creates formatted JSON */
+    public String createJson() {
+        Person person = new Person(this.name, this.profession, new ArrayList<>());
+        person.getPhoneNumbers().add(new PhoneNumber(this.phoneType1, this.number1));
+        person.getPhoneNumbers().add(new PhoneNumber(this.phoneType2, this.number2));
+
+        /* JSON-B config and Jsonb instance creation */
+        JsonbConfig config = new JsonbConfig()
+                .withFormatting(true);
+        Jsonb jsonb = JsonbBuilder.create(config);
+
+        /* Serialization to JSON */
+        this.jsonTextArea = jsonb.toJson(person);
+        
+        /* JSF navigation */
+        return "jsongenerated";
+    }
+    
+    /* Action method for form in jsongenerated.xhtml.
+     * Deserialize JSON from the textarea to Person object
+     * and fills these data to form. */
+    public String parseJson() {
+        clearFields();
+
+        /* JSON-B config and Jsonb instance creation */
+        Jsonb jsonb = JsonbBuilder.create(new JsonbConfig());
```

Comment:
Is it possible to create jsonb engine once per the application lifecycle? Creating it on every request is not the best practice.